lib/src/functions.dart

```diff
@@ -943,3 +950,24 @@ class Sgn extends DefaultFunction {
     throw UnimplementedError('Can not evaluate $name on $type yet.');
   }
 }
+
+class GenericFunction extends DefaultFunction {
+  /// Creates a genetic function with variable number of arguments.
+  ///
+  /// For example, to create min(2,23,4,5,2314,213,5,324):
+  ///
+  GenericFunction(String name, List<Expression> args, this.handler) : super._any(name, args);
+
+  dynamic handler;
+
+  @override
+  dynamic evaluate(EvaluationType type, ContextModel context) {
+    List<double> values = args.map<double>((v) => (v.value ?? context.getExpression(v.name)).evaluate(type, context)).toList();
+    if (type == EvaluationType.REAL) return handler(values);
+
+    throw UnimplementedError('Can not evaluate $name on $type yet.');
+  }
+
+  @override
+  Expression derive(String toVar) => Number(0);
```

Comment:
Maybe this should rather throw an exception since we can't be sure this is a constant value.